ynr/apps/uk_results/management/commands/import_2010_parl_results.py

```diff
@@ -0,0 +1,85 @@
+from django.core.management.base import BaseCommand
+from django.contrib.postgres.search import SearchQuery, SearchVector
+from slugify import slugify
+from candidates.models.popolo_extra import Ballot
+from django.db.models.functions import Length
+from popolo.models import Membership
+
+from uk_results.helpers import read_csv_from_url
+from uk_results import models
+
+
+class Command(BaseCommand):
+
+    EDGECASES = {
+        "ynys mon": "ynys môn",
+        "down south": "south down",
+        "down north": "north down",
+    }
+
+    def find_ballot(self, seat):
+        try:
+            return self.qs.get(post__label__iexact=seat)
+        except Ballot.DoesNotExist:
+            pass
+
+        matches = self.qs.filter(post__label__search=seat)
+        try:
+            return matches.get()
+        except Ballot.MultipleObjectsReturned:
+            return matches.annotate(label_len=Length("post__label")).get(
+                label_len=len(seat)
+            )
+
+    def find_candidacy(self, ballot, name):
+        # exact search first
+        try:
+            return ballot.membership_set.get(person__name__iexact=name)
+        except Membership.DoesNotExist:
+            pass
+
+        # fallback to a smarter searching
+        vector = SearchVector("person__name", "person__other_names__name")
```

Comment:
We actually have a vector column on the person model with this in already, so you should be able to do something like:

```
query = SearchQuery(name, search_type="raw", config="english")
ballot.membership_set.annotate(person__name_search_vector=query)
```

See here for more https://github.com/DemocracyClub/yournextrepresentative/blob/master/ynr/apps/search/utils.py

Reply:
(Updating that util to take in existing QuerySet would be useful too, but that's another conversation - there's also [a lot of problems with it!](https://github.com/DemocracyClub/yournextrepresentative/issues/1227))